crd-generator/apt/src/main/java/io/fabric8/crd/generator/apt/CustomResourceAnnotationProcessor.java

```diff
@@ -160,4 +162,34 @@ public void close() throws IOException {
       out.close();
     }
   }
+
+  private static Optional<ClassRef> getCustomResourceRef(TypeDef typeDef) {
+    return superClassStream(typeDef)
+      .filter(s -> s.getDefinition().getFullyQualifiedName().equals(CustomResource.class.getName()))
+      .findFirst();
+  }
+
+  private static Optional<ClassRef> getCustomResourceSpec(TypeDef typeDef) {
+    return getCustomResourceRef(typeDef)
+      .filter(c -> c.getArguments() != null && c.getArguments().size() == 2)
+      .map(c -> c.getArguments().get(0))
+      .filter(t -> t instanceof ClassRef)
+      .map(t -> (ClassRef) t);
+  }
+
+  private static Optional<ClassRef> getCustomResourceStatus(TypeDef typeDef) {
+    return getCustomResourceRef(typeDef)
+      .filter(c -> c.getArguments() != null && c.getArguments().size() == 2)
+      .map(c -> c.getArguments().get(1))
+      .filter(t -> t instanceof ClassRef)
+      .map(t -> (ClassRef) t);
+  }
+
+  public static Stream<ClassRef> superClassStream(TypeDef typeDef) {
```

Comment:
I was thinking about something like this but wasn't too sure how to go about it…